wagtail/admin/views/pages/revisions.py

```diff
@@ -39,6 +40,19 @@ def revisions_revert(request, page_id, revision_id):
         instance=revision_page, request=request, form=form
     )
 
+    if revision_page.live and revision_page.has_unpublished_changes:
+        # Page status needs to present the version of the page containing the correct live URL
+        page_for_status = page.specific
+    else:
+        page_for_status = revision_page
```

Comment:
Not quite confident of this, I just copied the logic from the edit view.

Reply:
I think we can just pass `page` to `PageSidePanels` instead. The purpose of this logic for the main edit view is to ensure that the URL linked from the 'Live' status button is based on the live version of the page rather than picking up changes to the slug that only exist as draft. I think the only reason for the `live` / `has_unpublished_changes` check was to potentially save a database query.

In this case, `revision_page` can be something from the distant past, so its `live` / `has_unpublished_changes` state is not relevant to the current live URL - instead, we always want to use `page.specific`, and we've already done that lookup as the very first line of the view.

Reply:
Ah, yes, I had a hunch that was the case. Thanks, I've changed this to always use `page.specific` instead.